### PR TITLE
Doors

### PR DESCRIPTION
**Pull request by LisanneW**
Doors and keys, only basic functions. No colors yet, and not tested.

### CODE REVIEW

File: MeesGame/Gameplay/GameObjects/Player.cs [hunk 1]
```diff
@@ -145,6 +154,16 @@ public override void HandleInput(InputHelper inputHelper)
                 nextAction = WEST;
             }
         }
+
+        public bool PlayerHasKey()
```
Comment (maxmeijer42) [inline MeesGame/Gameplay/GameObjects/Player.cs]: Ik zou dit hernoemen naar HasKey aangezien het een functie is van de Player klasse

File: MeesGame/Gameplay/GameObjects/Tile.cs [hunk 2]
```diff
@@ -41,7 +43,7 @@ public TileType TileType
 
     class FloorTile : Tile
     {
-        public FloorTile(int layer = 0, string id = "") : base("floorTile", TileType.Floor, layer, id)
+        public FloorTile(string assetName = "floorTile", TileType tt = TileType.Floor, int layer = 0, string id = "") : base(assetName, tt, layer, id)
```
Comment (maxmeijer42) [inline MeesGame/Gameplay/GameObjects/Tile.cs]: Deze constructor zou je protected kunnen maken zodat het type alleen door een subklasse kan worden gewijzigd

File: MeesGame/Gameplay/GameObjects/Tile.cs [hunk 3]
```diff
@@ -54,14 +56,27 @@ public override bool CanPlayerMoveHere(Player player)
...
+        public bool IsNextToSpecialTile()
+        {
+            return false;
         }
     }
 
     class WallTile : Tile
     {
-        public WallTile(int layer = 0, string id = "") : base("wall", TileType.Wall, layer, id)
+        public WallTile(string assetName = "fourWayWallTile", TileType tt = TileType.Wall, int layer = 0, string id = "") : base(assetName, tt, layer, id)
```
Comment (maxmeijer42) [inline MeesGame/Gameplay/GameObjects/Tile.cs]: Ik denk dat het beter is om de sprite naam te laten bepalen in de Tile klasses

Comment (jensj12) [reply MeesGame/Gameplay/GameObjects/Tile.cs]: Using "wall" crashes the game, so I think it's good to have a default that doesn't.

File: MeesGame/Gameplay/GameObjects/Tile.cs (hunk 2 above)
Comment (jensj12) [reply MeesGame/Gameplay/GameObjects/Tile.cs]: Then two constructors would be better. One public that doesn't allow changing the assetName and tileType and a protected one that does. There must be a public constructor in FloorTile.

Comment (maxmeijer42) [reply MeesGame/Gameplay/GameObjects/Tile.cs]: Yup, that's what I meant

File: MeesGame/Gameplay/GameObjects/DoorTile.cs [hunk 4]
```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using static MeesGame.PlayerAction;
```
Comment (jensj12) [inline MeesGame/Gameplay/GameObjects/DoorTile.cs:2]: Remove unnecessary usings.

File: MeesGame/Gameplay/GameObjects/Player.cs [hunk 5]
```diff
@@ -9,17 +9,21 @@ namespace MeesGame
     class Player : SpriteGameObject
     {
         protected Level level;
+        protected TileField tileField;
         protected int score = 0;
         protected Point location;
         protected TimeSpan lastActionTime;
         protected PlayerAction nextAction = NONE;
+        protected PlayerAction lastAction;
+        protected List<Tile> visitedTiles = new List<Tile>();
```
Comment (jensj12) [inline MeesGame/Gameplay/GameObjects/Player.cs:18]: I'd rather have a list of collected items instead of visited tiles.

**Review by jensj12 — CHANGES_REQUESTED** [review]

**Review by jensj12 — APPROVED** [review]

**Comment by jensj12** [comment]
It may or may not work, but as it contains important updates to the tile class and doors aren't used yet (in fact, they'll crash the game because of missing sprites), I'll merge it.

**Pull request merged**